dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/preheat/supplier/classStrategy/AbstractSchemaStrategy.java

```diff
@@ -83,7 +92,7 @@ public void add( TrackerPreheatParams params, List<List<String>> splitList, Trac
 
     @SuppressWarnings( "unchecked" )
     protected void queryForIdentifiableObjects( TrackerPreheat preheat, Schema schema, TrackerIdentifier identifier,
-        List<List<String>> splitList )
+        List<List<String>> splitList, Class<? extends PreheatMapper> mapper )
```

Comment:
Mappers generated by MapStruct can be regular Spring Beans, so if we use that approach we might avoid passing them around and we might rather just autowire them as dependency where needed

Reply:
Yes, this is a design choice. Since the mappers do not have any other dependency that requires injection and `MapStruct` has a very convenient method to fetch the mappers (via static method), we decided to keep things simple and avoid involving Spring. In the specific case you are commenting, Spring would have been actually more problematic that useful, because the `mapper` variable is specified in an annotation.